bottomless2/src/job.rs

```diff
@@ -1,14 +1,23 @@
+use std::sync::Arc;
+
 use crate::storage::Storage;
 use crate::Result;
-use crate::{NamespaceName, StoreSegmentRequest};
+use crate::StoreSegmentRequest;
+
+/// A request, with an id
+#[derive(Debug)]
+pub(crate) struct IndexedRequest<C> {
+    pub(crate) request: StoreSegmentRequest<C>,
```

Comment:
Why are we pub crating everything here? imo this is an anti-pattern

Reply:
I don't think it is, it's a primitive struct, has no behavior, no private state, no field validation and no stability guarantee, in other words, it's just a container. I would even argue that making those fields private is the anti-pattern here, as you'd make building the type more difficult, the code harder to refactor (adding a field means adding methods, and changing the constructor), opt-out of pattern matching, and overall make the code more verbose, for no benefit.